build-system/build.conf.js

```diff
@@ -26,9 +29,59 @@ const defaultPlugins = [
   require.resolve('./babel-plugins/babel-plugin-transform-version-call'),
 ];
 
+/**
+ * @return {Array<string|Object>} the minify-replace plugin options that can be
+ * pushed into the babel plugins array
+ */
+function getReplacePlugin() {
+  /**
+   * @param {string} defineStr the define flag to parse
+   * @return {Object} replacement options used by minify-replace plugin
+   */
+  function createReplacement(defineStr) {
+    const strSplit = defineStr.split('=');
+    const identifierName = strSplit[0];
+    // if no value is defined, set to true
+    const value = strSplit.length > 1 ? strSplit[1] === 'true' : true;
+
+    return {
+      identifierName,
+      replacement: {
+        type: 'booleanLiteral',
+        value,
+      },
+    };
+  }
+
+  const replacements = [];
+  // default each experiment flag constant to false
+  Object.keys(experimentsConfig).forEach(experiment => {
+    const experimentDefine = experimentsConfig[experiment]['define'];
+    if (experimentDefine) {
+      replacements.push(createReplacement(experimentDefine + '=false'));
+    }
+  });
+
+  // override define values from passed in flags
+  if (Array.isArray(argv.define)) {
+    argv.define.forEach(defineStr => {
+      replacements.push(createReplacement(defineStr));
+    });
+  } else if (argv.define) {
+    replacements.push(createReplacement(argv.define));
+  }
+
+  const replacePlugin = [
+    require.resolve('babel-plugin-minify-replace'),
+    {replacements},
+  ];
+
+  return replacePlugin;
+}
+
 module.exports = {
   plugins({isEsmBuild, isForTesting, isSinglePass}) {
-    let pluginsToApply = defaultPlugins;
+    let pluginsToApply = defaultPlugins.slice();
```

Comment:
I was getting an error locally where `babel-plugin-transform-amp-asserts` was duplicated in this array causing babel to error out. `pluginsToApply` is initially a reference to `defaultPlugins` and pushing this plugin caused subsequent uses of `defaultPlugins` to have this duplicate. Not sure why it was passing the PR build and failing locally but creating a copy seems to fix this.

Reply:
add `slice(0)` just to be explicit or do `[...defaultPlugins]`